### PR TITLE
megadriv.xml: Added eight homebrew entries.

### PR DESCRIPTION
**Pull request by 0kmg**
New working software list additions
-----------------------------------
Alter Ego Remastered [RetroSouls]
Gluf [RetroSouls]
Gravibots [RetroSouls]
Misplaced [RetroSouls]
Misplaced (Russian) [RetroSouls]
Old Towers (v1.2) [RetroSouls]
Old Towers (v1.1) [RetroSouls]
Yazzie [RetroSouls]

### CODE REVIEW

**Comment by 0kmg** [comment]
Charming little action puzzler games all. These are all available on cart but downloaded from what the developer released.

**Comment by mikehyrus** [comment]
Though there are other homebrews that exist through Piko Interactive like Custodian and Switchblade especially their English translated reproductions like Water Margin: A Tale of Clouds and Wind (Shui Hu Feng Yun Zhuan).  But that will be talked about some other time.

File: hash/megadriv.xml [hunk 1]
```diff
@@ -16970,6 +16992,17 @@ but dumps still have to be confirmed.
...
+	<software name="gravibots">
+		<description>Gravibots</description>
+		<year>2021</year>
+		<publisher>Mega Cat Studios</publisher>
+		<part name="cart" interface="megadriv_cart">
+			<dataarea name="rom" width="16" endianness="big" size="1441792">
+				<rom name="gravibots.bin" size="1441792" crc="0d66d022" sha1="71fa92a1d3cb61c1dc71a80e8029f98a988ba2f8"/>
+			</dataarea>
+		</part>
+	</software>
```
Comment (cuavas) [inline hash/megadriv.xml:17004]: I had a quick look at their page for this game, and it didn’t seem to have a download link, or at least not an obvious one.  Have they actually released this for free?  We generally don’t add original software to the software lists if it’s less than three years and hasn’t been released for free by the publisher.  (“Pay what you want” where zero is an acceptable amount but you need to confirm you’re paying zero to download doesn’t count as “released for free” for this purpose.)

Comment (0kmg) [reply hash/megadriv.xml:17004]: They've explicitly written "Grab a ROM for FREE or PWYW" but only on the 2019 ones. I assume that's their general stance, but I can't be sure. I'll redo this PR I guess, when I get a chance.

**Pull request closed without merging**